lib/rex/ui/text/dispatcher_shell.rb

```diff
@@ -210,6 +210,31 @@ def cmd_help(cmd=nil, *ignored)
           print(File.read(File.join(docs_dir + '.md')))
         end
       end
+
+      # smb, mysql, mssql and postgresql session types make use of `shell.session` which type can then
+      # be called against (`shell.session.type`)
+      #
+      # Meterpreter seems to make use of `shell.client` which type can then be called against (`shell.client.type`)
+      #
+      # checking this to not raise a no methods error
+      unless defined?(shell.session).nil?
+        # List of session types we want to list compatible modules for.
+        # Wont include Meterpreter and shell as that would be too long a list.
+        session_types_to_display_compatible_modules_for = %w[smb mysql mssql postgresql]
+
+        if session_types_to_display_compatible_modules_for.include?(shell.session.type)
+          # Use the built in search command functionality to get a list of search results
+          search_term = "session_type:#{shell.session.type}"
+          search_params = Msf::Modules::Metadata::Search.parse_search_string(search_term)
+          search_results = Msf::Modules::Metadata::Cache.instance.find(search_params)
+          # Filter those down to just the modules fullname
+          session_compatible_modules = search_results.flat_map(&:fullname)
+          print_line('This session also works with the following modules:')
+          print_line
+          session_compatible_modules.each { |session_compatible_module| print_line("  #{session_compatible_module}") }
+          print_line
+        end
+      end
```

Comment:
Are these changes meant to be here?

i.e. Is this session list meant to be appearing for _every_ help command run? 👀 

```
PostgreSQL @ 127.0.0.1:9000 (template1) > help query
Usage: query

Run a raw SQL query on the target.

Examples:

    query SELECT user;
    query SELECT version();
    query SELECT * FROM pg_catalog.pg_tables;

This session also works with the following modules:

  auxiliary/admin/postgres/postgres_readfile
  auxiliary/admin/postgres/postgres_sql
  auxiliary/scanner/postgres/postgres_hashdump
  auxiliary/scanner/postgres/postgres_schemadump
  auxiliary/scanner/postgres/postgres_version
  exploit/linux/postgres/postgres_payload
  exploit/multi/postgres/postgres_copy_from_program_cmd_exec
  exploit/multi/postgres/postgres_createlang
  exploit/windows/postgres/postgres_payload

```

Reply:
Nope. I'll update that to just be with the `help` menu not when args are passed 😄